selfhost/posix_fs.jakt

```diff
@@ -35,3 +47,49 @@ function current_directory() throws -> String {
     b.append_c_string(buf)
     return b.to_string()
 }
+
+class DirectoryIterator implements(ThrowingIterable<(String, bool)>) {
+    path: String
+    dir_fd: raw DIR
+
+    public function next(mut this) throws -> (String, bool)? {
+        let file = readdir(.dir_fd)
+
+        if file == null<dirent>() {
+            closedir(.dir_fd)
+
+            return None
+        }
+
+        mut builder = StringBuilder::create()
+        mut is_directory = false
+
+        unsafe {
+            builder.append_c_string((*file).d_name)
+
+            if (*file).d_type == 4 {
```

Comment:
Please leave a comment here about this constant; we should really think up a way to import #defines sooner rather than later

Reply:
Do we want constants or similar in the language? This will probably come up a lot with serenity interop.

Might be clearer atm just to do something like:
```
cpp {
    "if (*(file).d_type == DT_DIR)"
    "is_directory = true"
}
```

Reply:
I think I'm leaning towards "yes, we want constants", but extern-importing them is pretty much required :shrug:

Reply:
Both of the constant cases used cpp and added a fixme for extern importing them.